### PR TITLE
LinkageChecker.Builder to eliminate nulls in LinkageChecker creation

### LINKED ISSUE #1321: Linkage Checker Builder?
Currently we have overloaded static factory `LinkageChecker.create`:

```
Bom baseline = ...
LinkageChecker.create(baseline, null).findSymbolProblems();
```

```
LinkageChecker linkageChecker = LinkageChecker.create(classpath, entryPoints, null);
```

```
LinkageChecker linkageChecker = LinkageChecker.create(classPath);
```

# LinkageChecker.Builder

Builder pattern may be suitable here:

```
LinkageChecker linkageChecker = LinkageChecker.builder(classPath)
    .entryPoint(entryPointJars)
    .exclusionFile(exclusionFile)
    .build();
```

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #1321

### CODE REVIEW

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ public static void main(String[] arguments)
     logger.info("BOM Coordinates: " + latestBomCoordinates);
     Bom baseline = RepositoryUtility.readBom(latestBomCoordinates);
     ImmutableSet<SymbolProblem> problemsInBaseline =
-        LinkageChecker.create(baseline, null).findSymbolProblems().keySet();
+        LinkageChecker.builder(baseline).build().findSymbolProblems().keySet();
```
Comment (suztomo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: The nulls in the static factory methods have been removed.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java [hunk 2]
```diff
@@ -65,56 +65,12 @@ public ClassReferenceGraph getClassReferenceGraph() {
...
-        classReferenceGraph,
-        ExcludedErrors.create(exclusionFile));
-  }
-
-  public static LinkageChecker create(Bom bom) throws IOException {
-    return create(bom, null);
-  }
-
-  public static LinkageChecker create(Bom bom, Path exclusionFile) throws IOException {
-    // duplicate code from DashboardMain follows. We need to refactor to extract this.
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java]: These overloaded factory methods have been removed.

**Comment by suztomo** [comment]
Merged with master.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 package com.google.cloud.tools.opensource.classpath;
 
-import static com.google.cloud.tools.opensource.classpath.LinkageCheckerTest.resolvePaths;
+import static com.google.cloud.tools.opensource.classpath.LinkageCheckerTest.resolve;
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java:19]: This shouldn't be a static import, and probably one test should not be depending on another like this.

**Comment by suztomo** [comment]
Now I think we need to have `LinkageCheckRequest` and its `Builder`. I'll revise that.

**Pull request closed without merging**